governance/README.md

```diff
@@ -0,0 +1,89 @@
+the new program data and overwriting of the existing Program account's data with it is handled in the background for you
+by the Solana program deploy cli command.
+However, in order for Governance to be useful, Governance now needs this authority.
+
+### Proposal accounts
+
+A Proposal is an instance of a Governance created to vote on and execute given set of changes.
+It is created by someone (Proposal Admin) and tied to a given Governance account
+and has a set of executable commands to it, a name and a description.
+It goes through various states (draft, voting, executing) and users can vote on it
+if they have relevant Governance or Council tokens.
+It's rules are determined by the Governance account that it is tied to, and when it executes,
+it is only eligible to use the [Program Derived Address](https://docs.solana.com/developing/programming-model/calling-between-programs#program-derived-addresses)
+authority given by the Governance account.
+So a Proposal for Sushi cannot for instance upgrade the Program for Uniswap.
+
+When a Proposal is created by a user then the user becomes Proposal Admin and receives an Admin an Signatory token.
+With this power the Admin can add other Signatories to the Proposal.
+These Signatories can then add commands to the Proposal and/or sign off on the Proposal.
+Once all Signatories have signed off on the Proposal the Proposal leaves Draft state and enters Voting state.
+Voting state lasts as long as the Governance has it configured to last, and during this time
+people holding Governance (or Council) tokens may vote on the Proposal.
+Once the Proposal is "tipped" it either enters the Defeated or Executing state.
+If Executed, it enters Completed state once all commands have been run.
+
+A command can be run by any one at any time after the `delay_slots` length has transpired on the given command.
+
+### CustomSingleSignerInstruction
+
+We only support one kind of executable command right now, and this is the CustomSingleSignerInstruction type.
+A Proposal can have a certain number of these, and they run independently of each other.
+These contain the actual data for a command, and how long after the voting phase a user must wait before they can be executed.
+
+### Voting Dynamics
+
+Each Proposal that gets created creates a holding account for all source tokens that voters will deposit in order to vote.
+To vote, one needs to deposit the source token for a given Proposal. One may deposit let's say 1 Uniswap and get 1 vote.
+The source tokens must be provided to the Vote instruction and are put into the Source Token Holding account.
+
+Once the vote completes, the source tokens can be withdrawn from the holding account of the Proposal.
+It's also possible to change your mind and withdraw your tokes before voting is completed but in such a case you vote
+won't count towards the final outcome of the Proposal.
```

Comment:
What happens if two proposals are up for vote at the same time?

Reply:
We are changing the mechanism of voting based on the PR feedback and README hasn't been updated accordingly yet. In the final version we won't have this restriction.